contrib/babelfishpg_common/src/varchar.c

```diff
@@ -1188,6 +1191,84 @@ fixeddecimal2bpchar(PG_FUNCTION_ARGS)
 #undef FIXEDDECIMAL_2_VARCHAR_MULTIPLIER
 #undef FIXEDDECIMAL_2_VARCHAR_SCALE
 
+Datum
+float82varchar(PG_FUNCTION_ARGS)
+{
+	float8 num = PG_GETARG_FLOAT8(0);
+	int32 typmod = PG_GETARG_INT32(1);
+	int maxlen = (typmod == -1) ? 0 : (typmod - VARHDRSZ);
+	Datum res;
+	char	   *ascii = (char *) palloc(32);
```

Comment:
lets use palloc0

Reply:
and also add comment on this magic number 32

Reply:
Fixed